commit 134cdfa2b0ad3eb883f334028b5a27e8ddf63a0f
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2019-05-07T16:15:58+00:00 (5 months ago)
    
    TRITON-1623 vmapi migration test has intermittent 'bandwidth progress event was seen' failure

diff --git a/lib/vm-migration/migrate.js b/lib/vm-migration/migrate.js
index db92d27..f1209de 100644
--- a/lib/vm-migration/migrate.js
+++ b/lib/vm-migration/migrate.js
@@ -801,6 +801,12 @@ function watchVmMigration(req, res, next) {
         // running).
         if (!MIGRATION_RECORD_FOR_VM_UUID[vmUuid] ||
                 MIGRATION_RECORD_FOR_VM_UUID[vmUuid].state !== 'running') {
+            // Send out any pending progress events first.
+            if (MIGRATION_PROGRESS_FOR_VM_UUID[vmUuid]) {
+                res.write(JSON.stringify(MIGRATION_PROGRESS_FOR_VM_UUID[vmUuid])
+                    + '\n');
+            }
+            // Send the end event.
             res.write(JSON.stringify(
                 {
                     type: 'end',
diff --git a/package.json b/package.json
index c1c484d..b6fde8d 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "vmapi",
   "description": "VMs API",
-  "version": "9.8.5",
+  "version": "9.8.6",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
